### PR TITLE
[Spec] Light client proofs

### PR DESCRIPTION
**Pull request by bowenwang1996**
Add spec on merkle proofs and light client proofs for execution outcomes. Requires https://github.com/nearprotocol/nearcore/pull/2737

### CODE REVIEW

File: specs/ChainSpec/LightClient.md [hunk 1]
```diff
@@ -169,3 +239,14 @@ The RPC returns the `LightClientBlock` for the block as far into the future from
...
+
+### Light Client Proof
+
+The following rpc end-point returns `RpcLightClientExecutionProofResponse` that a light client needs for verifying execution outcomes:
+
+```
+http post http://127.0.0.1:3030/ jsonrpc=2.0 method=EXPERIMENTAL_light_client_proof params:="{"type": <proof_type>, "id": <id>, "sender": <sender>, "light_client_head": <light_client_head>}" id="dontcare"
+```
+
+Note that the above command shows the case for querying proof for transaction outcome. For receipt outcome, "sender" should
```
Comment (SkidanovAlex) [inline specs/ChainSpec/LightClient.md:251]: Put a whole example for receipt?

File: specs/ChainSpec/LightClient.md [hunk 2]
```diff
@@ -156,7 +163,70 @@ The sum of all the stakes of `next_bps` in the previous epoch is `total_stake` r
...
 The signatures in the `LightClientBlockView::approvals_after_next` are signatures on `approval_message`. The  `i`-th signature in `approvals_after_next`, if present, must validate against the `i`-th public key in `next_bps` from the previous epoch. `approvals_after_next` can contain fewer elements than `next_bps` in the previous epoch.
 
-## RPC end-point
+## Proof Verification
+
+[transaction-outcome-proof](Transaction Outcome Proofs)
+### Transaction Outcome Proofs
+
+To verify that a transaction or receipt happens on chain, a light client can request a proof through rpc by providing `id`, which is of type
+```rust
```
Comment (SkidanovAlex) [inline specs/ChainSpec/LightClient.md:172]: Does it actually render properly without an extra line break before "```"?

File: specs/ChainSpec/LightClient.md [hunk 3]
```diff
@@ -156,7 +163,70 @@ The sum of all the stakes of `next_bps` in the previous epoch is `total_stake` r
...
+pub enum TransactionOrReceiptId {
+    Transaction { hash: CryptoHash, sender: AccountId },
+    Receipt { id: CryptoHash, receiver: AccountId },
+}
+```
+and the block hash of light client head. The rpc will return the following struct
+```rust
+pub struct RpcLightClientExecutionProofResponse {
+    /// Proof of execution outcome
+    pub outcome_proof: ExecutionOutcomeWithIdView,
```
Comment (SkidanovAlex) [inline specs/ChainSpec/LightClient.md:182]: Is it defined somewhere?

File: specs/ChainSpec/LightClient.md (hunk 2 above)
Comment (bowenwang1996) [reply specs/ChainSpec/LightClient.md:172]: I think so

**Pull request merged**